cmd/edit/cluster/cmd.go

```diff
@@ -1142,3 +1189,163 @@ func confirmMigration() (bool, error) {
+	}
+
+	// If updating role ARN with AutoNode already enabled, maintain enabled state
+	if roleArnChanged && currentAutoNodeEnabled && !autonodeChanged {
+		mode = "enabled"
+	}
+
+	// Confirm with user
+	if mode != "" || role != "" {
+		confirmAutoNode(r, mode, role, currentAutoNodeEnabled, currentRoleArn)
+	}
+
+	return mode, role, nil
+}
+
+func confirmAutoNode(r *rosa.Runtime, mode string, roleArn string, currentEnabled bool, currentRoleArn string) {
+	if mode == "enabled" && !currentEnabled {
+		r.Reporter.Warnf("You are choosing to enable AutoNode for automatic node management")
+		if !confirm.Confirm("enable AutoNode for cluster with the provided role arn '%s'", roleArn) {
+			os.Exit(0)
+		}
+	} else if roleArn != "" && currentRoleArn != "" && roleArn != currentRoleArn {
+		r.Reporter.Warnf("You are choosing to update the AutoNode IAM role ARN")
+		if !confirm.Confirm("update AutoNode IAM role ARN from '%s' to '%s'", currentRoleArn, roleArn) {
+			os.Exit(0)
+		}
+	}
+}
+
+func autonodeInteractivePrompt(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster) (
+	mode string, roleArn string, err error) {
+
+	// Only prompt if AutoNode is not already enabled
+	currentAutoNodeEnabled := autonodeExists(cluster)
+	if currentAutoNodeEnabled {
+		// AutoNode is already enabled, optionally allow updating the role ARN
+		currentRoleArn := autonodeRoleExists(cluster)
+
+		updateRoleArn, err := interactive.GetBool(interactive.Input{
+			Question: fmt.Sprintf("Update AutoNode IAM Role ARN (current: %s)", currentRoleArn),
+			Default:  false,
+			Required: false,
+		})
+		if err != nil {
+			return "", "", fmt.Errorf("Expected a valid value: %s", err)
+		}
+
+		if updateRoleArn {
+			roleArnValue, err := interactive.GetString(interactive.Input{
+				Question: "New AutoNode IAM Role ARN",
+				Help:     cmd.Flags().Lookup("autonode-iam-role-arn").Usage,
+				Default:  currentRoleArn,
+				Required: true,
+				Validators: []interactive.Validator{
+					interactive.RegExp(aws.RoleArnRE.String()),
+				},
+			})
+			if err != nil {
+				return "", "", fmt.Errorf("Expected a valid value for autonode-iam-role-arn: %s", err)
+			}
+			// Maintain enabled state when updating role
+			return "enabled", roleArnValue, nil
+		}
+		return "", "", nil
+	}
+
+	// AutoNode is not enabled, prompt to enable
+	enableAutoNode, err := interactive.GetBool(interactive.Input{
+		Question: "Enable AutoNode for automatic node management",
+		Default:  false,
+		Required: false,
+	})
+	if err != nil {
+		return "", "", fmt.Errorf("Expected a valid value: %s", err)
+	}
+
+	if !enableAutoNode {
+		return "", "", nil
+	}
+
+	r.Reporter.Infof("AutoNode requires an IAM role with appropriate permissions for node management.")
+	r.Reporter.Infof("Please ensure the IAM role is created before proceeding.")
+
+	roleArnValue, err := interactive.GetString(interactive.Input{
+		Question: "AutoNode IAM Role ARN",
```

Comment:
I think we could get away with 2 prompts rather than 4, WDYT? Even if we want the different prompt messages provided in the above prompts, we can do so while only having 2 prompts in the code

Reply:
You're right, the 2 prompts are more concise, fixed on the last commit